src/client/protocol.c

```diff
@@ -103,7 +103,8 @@ static int peekUint64(struct cursor cursor, uint64_t *val)
 	if (cursor.cap < 8) {
 		return DQLITE_CLIENT_PROTO_ERROR;
 	}
-	*val = ByteFlipLe64(*(uint64_t *)cursor.p);
```

Comment:
Where is the arithmetic on a void pointer happening in this case (and similar cases) ?

Reply:
There is no void pointer arithmetic going on, the problem is just that once `cursor.p` is no longer a void pointer the compiler starts to emit a warning about alignment on the cast here. Using memcpy instead gets rid of that.

Reply:
`char *` has no alignment requirements (and can alias with anything) so changing the type can't create any alignment problems.

Reply:
But those situations would already have UB with `void *` right?

Reply:
yeah, didn't know void* and char* had same alignment requirements.